src/bentoml/_internal/server/http/instruments.py

```diff
@@ -124,6 +125,21 @@ async def __call__(
             await response(scope, receive, send)
             return
 
+        # NOTE: Exemplars are reference to data outside
+        # of the metrics set. This is often use to link
+        # traces to given metrics.
+        exemplar: dict[str, str] = dict(
+            filter(
+                lambda tup: tup[1] is not None,
+                {
+                    "trace_id": "%s" % trace_context.trace_id,
+                    "span_id": "%s" % trace_context.span_id,
+                    "request_id": "%s" % trace_context.request_id,
+                    "sampled": "%s" % trace_context.sampled,
+                }.items(),
+            )
+        )
```

Comment:
I'm somewhat unsure what's happening here, it seems like you're casting all the values into strings with `%` instead of `str`, and then filtering out `None`, but nothing should be `None`, since `"%s" % None` should be `'None'`.

```suggestion
        exemplar: dict[str, str] = {
            key: val for key, val in [
                "trace_id": "%s" % trace_context.trace_id,
                 "span_id": "%s" % trace_context.span_id,
                 "request_id": "%s" % trace_context.request_id,
                 "sampled": "%s" % trace_context.sampled,
            ] if val is not None
        }
```